### DISCUSSION TITLE
ie 11 support

### DISCUSSION THREAD

**Discussion by devamigo** [soofstad/react-oauth2-pkce#127] [Q&A]
Not sure this has been tested in ie11 , but it erred. 

pckeUtils.ts: line 32 has 2 issues for ie 11.

1. windows.crypto may casue issue?  (windows.crypto || window.msCrypto) should be ok?
2. once that is changed, msCrypto actually returns an event so having that in a Promise and the await could resolve  it?

(IE11 uses an old standard of the [Web Crypto AP](https://developer.mozilla.org/en-US/docs/Web/API/Web_Crypto_API)I, the [digest](https://developer.mozilla.org/en-US/docs/Web/API/SubtleCrypto/digest) method does not return a promise which happens in all other browsers, instead, it returns a CryptoOperation. There is hardly any reference on the internet about this mysterial operation, as per [OAUTH PKCE: Generate code_verifier and code_challenge in IE11 and modern browsers](https://coolgk.medium.com/oauth-pkce-generate-code-verifier-and-code-challenge-in-ie11-and-modern-browsers-e0b8864956ed))

May be a support for 1e11 could be beneficial?

**Comment by sebastianvitterso** [comment]
Support for IE11 has been dropped by Microsoft themselves, and [customers are encouraged to move to a more modern browser](https://learn.microsoft.com/en-us/lifecycle/announcements/internet-explorer-11-end-of-support). It is no longer considered secure to venture out onto the internet using IE11, as no new security patches will ever again be made for the browser.

If your users use IE11, you should **implore** them to migrate over to Edge, Chrome, Firefox, Safari or any other more modern (continuously updated) browser!

---

I don't personally think we should support IE11 (as you might have gathered from my reasoning above), but I'm interested in hearing _why_ you need the library to support IE11. Do you have a user base using IE11? Are you developing for an internal system built upon specificities of IE11?

If there's some reason we haven't thought of, we might consider it (as I agree; the required change is quite small).
